openfire-oidc/src/main/java/com/gst/openfire/oidc/auth/OidcTokenValidator.java

```diff
@@ -34,47 +34,21 @@
 public class OidcTokenValidator {
 
     private static Logger logger = LoggerFactory.getLogger(OidcTokenValidator.class);
-    private JiveProperties jiveProperties;
+    private static Key keycloakPublicKey;
 
-    public OidcTokenValidator(JiveProperties jiveProperties) {
-        logger.info("Init keycloak token validator");
-        this.jiveProperties = jiveProperties;
-    }
-
-    public OidcTokenValidator() {
-    }
-
-    public JwtClaims verifyToken(String token) throws MalformedClaimException, InvalidJwtException,
-            NoSuchAlgorithmException, InvalidKeySpecException, JoseException, IOException {
-
-        //TODO: Read the auth server from the configuration instead of the token
-        String url = getIssuerFromToken(token);
-        if (url == null) {
-            throw new IllegalStateException("no issuer found in token");
+    static {
+        try {
+            keycloakPublicKey = getKeycloakPublicKey(JiveProperties.getInstance().get("oidc.auth.url"));
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Không bao giờ dùng in stack trace kiểu này, cái này chỉ dùng để code chơi chứ ko dùng trong product